lib/Scheduling/SimplexSchedulers.cpp

```diff
@@ -923,21 +928,28 @@ LogicalResult SharedOperatorsSimplexScheduler::schedule() {
                             getStartTime(startTimeVariables[b]);
                    });
 
-  // Store the number of operations using an operator type in a particular time
+  // Store the number of operations using a resource type in a particular time
   // step.
-  SmallDenseMap<Problem::OperatorType, SmallDenseMap<unsigned, unsigned>>
+  SmallDenseMap<Problem::ResourceType, SmallDenseMap<unsigned, unsigned>>
       reservationTable;
 
   for (auto *op : limitedOps) {
-    auto opr = *prob.getLinkedOperatorType(op);
-    unsigned limit = prob.getLimit(opr).value_or(0);
+    auto maybeRsrcs = prob.getLinkedResourceTypes(op);
+    assert(maybeRsrcs && "Limited operation must have linked resource types");
+
+    auto &rsrcs = *maybeRsrcs;
+    assert(rsrcs.size() == 1 &&
+           "Multi-resource operations are not yet supported by this scheduler");
```

Comment:
Please address this in a follow-up PR. We should have at least one complete, reference scheduler implementation for each problem defined in-tree. 🙏

Reply:
Yes, no problem - this is actually an already ongoing patch!